### PR TITLE
Update `black` documentation

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

In the Dev telecon there was a discussion of the documentation for `black`. This PR tweaks the docs for `black` based on that discussion.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/development/codeguide.rst [hunk 1]
```diff
@@ -148,6 +150,11 @@ Coding Style/Conventions
       * or one can add a single ``# fmt: skip`` comment to the end of a line to
         disable ``black`` formatting for that line.
 
+      .. note::
+        ``black`` can also be manipulated into one-lining (assuming the statement
+        fits inside the 88 character line length limit) or multi-lining a statement
+        by adding a trailing comma to the end of the line.
```
Comment (dhomeier) [inline docs/development/codeguide.rst:156]: I know the use case this refers to, but I am not sure I'd understand it from the description (actually I recall it is enabling more "matrix-like" formatting, but what exactly is the effect – allowing > 1 item per line in a multi-line block?).
I think @eteq's comment on [adding a `+`](https://astropy.slack.com/archives/C04A0DV3680/p1668201774735609) might also be useful here.

Comment (taldcroft) [reply docs/development/codeguide.rst:156]: Maybe something like this:
```
You can communicate to ``black`` that a list or array should be formatted as one item
per line using the `magic trailing comma 
<https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma>`_.
This frequently helps with formatting arrays.
```

File: docs/development/codeguide.rst [hunk 2]
```diff
@@ -148,6 +150,13 @@ Coding Style/Conventions
       * or one can add a single ``# fmt: skip`` comment to the end of a line to
         disable ``black`` formatting for that line.
 
+      .. note::
+        You can communicate to ``black`` that a list or array should be formatted
+        as one item per line using the
+        `magic trailing comma <https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma>`_.
+        This adding/removing this comma frequently helps with formatting "list
+        like" structures such as function input arguments or arrays.
```
Comment (WilliamJamieson) [inline docs/development/codeguide.rst:158]: @dhomeier, @taldcroft, and/or @pllim  does this seem better?

[the remark this answers is not in the document]
Comment (dhomeier) [reply docs/development/codeguide.rst:158]: +1, only I'd suggest "one row/subarray per line" ("item" sounds more like a single element to me),
and "sill allowing" -> "still allowing".

**Review by mhvk — APPROVED** [review]
Yes, nice addition! It took me a bit to realize that this trailing comma was relevant (also that one can remove it to get a one-liner in some cases).

File: docs/development/codeguide.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (eerovaher) [reply docs/development/codeguide.rst]: > Also I am still curious what @eteq's note on
> 
> > if I use some of the tricks like adding a + to get explicit matrices to line up
> 
> was about. upside_down_face

It is about vertical alignment of matrix elements, see e.g. https://github.com/astropy/astropy/blob/bd0ec6f5681ee4b79c1b742c93e0a3f31167cde0/astropy/coordinates/builtin_frames/fk4_fk5_transforms.py#L16-L22

File: docs/development/codeguide.rst [hunk 3]
```diff
@@ -151,6 +153,19 @@ Coding Style/Conventions
     This should be done sparingly, and only
     when ``black`` produces undesirable formatting.
 
+      .. note::
+        When a list or array should be formatted as one item per line then this is best
+        achieved by using the
+        `magic trailing comma <https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma>`_.
+        This is frequently sufficient for formatting matrices while sill allowing
```
Comment (dhomeier) [inline docs/development/codeguide.rst:160]:
```suggestion
        This is frequently sufficient for formatting matrices while still allowing
```

File: docs/development/codeguide.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhomeier) [reply docs/development/codeguide.rst]: Nice. But maybe elementary, not sure if it warrants extra mentioning.

**Comment by mhvk** [comment]
Had a quick look and I'm still happy with this.

File: docs/development/codeguide.rst [hunk 4]
```diff
@@ -151,6 +153,19 @@ Coding Style/Conventions
     This should be done sparingly, and only
     when ``black`` produces undesirable formatting.
 
+      .. note::
+        When a list or array should be formatted as one item per line then this is best
+        achieved by using the
+        `magic trailing comma <https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma>`_.
+        This is frequently sufficient for formatting matrices while still allowing
+        ``black`` to check the code::
```
Comment (dhomeier) [inline docs/development/codeguide.rst:161]:
```suggestion
        This is frequently sufficient for keeping matrices formatted as one row
        per line while still allowing ``black`` to check the code::
```
The only thing I've been stumbling on is that the trailing comma does not really trigger any new formatting here, it just tells black to preserve the existing one-per-line format – perhaps this is making it clearer.

**Review by dhomeier — CHANGES_REQUESTED** [review]
Only one comment to consider, otherwise looks good, thanks!

**Review by dhomeier — APPROVED** [review]

**Review by eerovaher — COMMENTED** [review]
I like the changes, but I'd recommend squashing the commits.

**Comment by WilliamJamieson** [comment]
> I like the changes, but I'd recommend squashing the commits.

Done

**Review by eerovaher — APPROVED** [review]

**Pull request merged**